### ISSUE TITLE
Paramiko uses 100% CPU with ProxyCommand when "ssh -W" was killed

### ISSUE THREAD

**Issue by pysnow530** [paramiko/paramiko#1277]
Hi there,

When I use ProxyCommand(), paramiko will generate the proxy process. If the proxy process was killed before paramiko connection closed, the cpu will up to 100%.

Here is the information and reproduce steps.

```
$ pip freeze | grep paramiko
paramiko==2.1.2
```

```
$ cat paramiko_proxy.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
import time

import paramiko
from paramiko import client

proxy = paramiko.ProxyCommand('ssh -q -W 127.0.0.1:22 127.0.0.1 2>/dev/null')

client = client.SSHClient()
client.set_missing_host_key_policy(paramiko.AutoAddPolicy())
client.connect('localhost', sock=proxy)
stdin, stdout, stderr = client.exec_command('whoami')
print stdout.read()

time.sleep(10000)

# ---> when sleeping, kill the "ssh -W" process, the cpu will up to 100%.
```

**Comment by pysnow530** [comment]
Maybe some relationship with #413

**Comment by ploxiln** [comment]
possible fix is in #1180
